c++/src/kj/exception.h

```diff
@@ -135,8 +135,14 @@ class Exception {
   // Adds the location that called this method to the stack trace.
 
   using DetailTypeId = unsigned long long;
+  struct Detail {
+    DetailTypeId id;
+    kj::Array<byte> value;
+  };
 
   kj::Maybe<kj::ArrayPtr<const byte>> getDetail(DetailTypeId typeId) const;
+  kj::ArrayPtr<const Detail> getDetails() const;
+  bool hasDetails() const;
```

Comment:
`hasDetails()` isn't strictly necessary here. Someone could just check `getDetails().size() != 0`, or have `getDetail()` return a `kj::Maybe<kj::ArrayPtr<const Detail>>` (but I prefer just the size check).

Reply:
Done, I was over complicating that.

Reply:
Helps if I push up the change :D